### ISSUE TITLE
[Updated] YouTube Plus might end with the upcoming YouTube Material Layout

### ISSUE THREAD

**Issue by ParticleCore** [ParticleCore/Particle#448]
EDIT2: **The new version for the YouTube Material Layout has been published here:** https://github.com/ParticleCore/Particle/issues/493

Original text kept below

---

EDIT: **YT+ for Material Layout is under works, read more about it here**: https://github.com/ParticleCore/Particle/issues/448#issuecomment-293663902

Original text kept below

---

For the better part of almost one year YouTube started working on its new Material based Layout: http://www.theverge.com/2016/5/3/11576016/youtube-material-design-test

Right from the start trying to work on the new layout proved to be difficult, with constant changes every week breaking any kind of progress made. For this reason the work on the YT+ Material compatible version stopped until the new layout became more stable. Until about four months ago I resumed the work and started studying the new layout and immediately noticed big challenges: despite the similarities, the Firefox and Chrome Material layout versions are not the same. YouTube is now using Polymer for the new layout, but due to limitations on Firefox (or just Google trying to push their standards?) this meant that there were two different structures to work with, meaning that YT+ would require a lot more code to avoid having two different versions, each one specific to each browser.

Polymer works as a template & recycle structure and, although Firefox presented very few challenges, Chrome, on the other hand, was a completely different situation. Due to the way the Material layout is structured, the majority of the elements necessary to work are not ready when the pages are presented, instead they are constructed once the information has finished loading. This causes a lot of problems since there are quite a few elements YT+ needs to work, which are not available when it needs to, namely the navigation bar and the video thumbnails.

While I have been able to work through these challenges, the level of difficulty and complexity have become borderline unreasonable. Although going a different route and wait for the page to finish "constructing" would solve the biggest headaches, another problem arises. Since the Polymer is recycling elements, all removed blacklist videos would cause incorrect videos to not show (during page refresh or changes) or remove incorrect videos. This meant that at least one big feature would no longer work.

The problems don't end here, since this layout hasn't been launched officially, any progress made to this date could become broken when the launch becomes official. Font scaling is also another huge problem which now appears to be different, depending on the users pixel density, and the worst part of all of these problems is that the Material layout is terribly heavy. Memory consumption isn't too bad, but the navigation in vanilla (without YT+) is sluggish and every time a new page is loaded the entire browser locks up when YouTube "constructs" the new page with the received information. Any experience so far in this layout, with and without YT+, has been terrible and this hasn't changed since the layout was first tested.

At this exact moment the YT+ Material version source code is opened on the other half of the screen, stopped at the welcome message and I have started to dread resuming work. The joy of working on YT+ has been dwindling since work on the Material version began and it reached the bottom today.

This isn't fun for me anymore, my free time has been very scarce and might become nonexistent in less than 15 days. With all of the above problems culminating I have decided that YT+ will very likely die once the Material Layout goes live. I wish I had more time and resources to dedicate to this project full time, but simply put the donations aren't enough (about $8/month) and I have to focus on more important things in my life.

Do not take me wrong, I appreciate every donation made, even the few who donated $0.30 which were totally absorbed by the transaction commissions, I didn't even want to ask for donations, some even criticized me because I didn't publicize the extension, but these extras, from those that could share, always helped. And no, I am not forgetting the very valuable users input, which helped fix and improve a lot of problems in YT+.

It has been [2 years and 29 days](https://github.com/ParticleCore/Particle/commit/145e9ce5405d0ef362ac230d26c6c0c30c24a591), a total of [176 versions](https://github.com/ParticleCore/Particle/releases) have been released so far and a total of over 60,000 users have been boarding this experience. When it started it was nothing but a script for myself to enjoy YouTube better, it had no settings menu, no structure, just a bunch of code garbled up. This project helped me learn a ton of new things and provided me with an incredible perspective of the developers' side. Light, useful and fast were always my priorities for this extension.

I will continue maintaining the current vanilla version of YT+ and will keep an eye out for the Material layout to see if, perhaps, something changes in me again, however don't get your hopes up, I am tired and my life is not easy. On a different note, I might also go forward with the [dedicated Blacklist and AdBlocker for YouTube Material](https://github.com/ParticleCore/Particle/issues/351), these were the two features I worked the most while trying to make YT+ compatible with the new layout and they appear to be relatively simple to maintain as individual extensions and extremely useful, at least to me.

Again, everyone is free to fork the project, I only ask that [the license](https://github.com/ParticleCore/Particle/blob/master/LICENSE) is respected. It is small and easy to understand, it shouldn't take more than 1 minute to read.

I'd like to thank again the author of YouTube Center, [Jeppe Rune Mortensen](https://github.com/YePpHa), whose work was the inspiration for the creation of YouTube Plus.

It has been fun.

**Comment by Keefaetana** [comment]
I'm sad to see you go but I certainly understand. Thank you for everything so far. Your work was, is and will be appreciated.

**Comment by Eisys** [comment]
This sucks. Worst news I've heard in a long time. Especially since I can not use Youtube without this script. Really couldn't live without 'floating player', 'shortcuts always active' and 'disable HFR' for example. 
It does everything I need. And there is NO other alternative that comes close to being as good as YT+.

Google, once again, fucking everything good up. Last year they made alot of changes that pissed off youtubers and now with this new idiotic layout, they piss off the viewers.

I very much hope that there is some way for you to keep it updated after material goes live or perhaps Google realise how awfully it runs and makes it better themselves.

Since you mentioned Firefox and Chrome being so different, any chance of just focusing on one browser? I think I'd be willing to switch just to have YT+, if Firefox's version is easier to work with for example.

But in case, thank you so very much for all that you've done with YT+. It has been amazing.

**Comment by q1k** [comment]
I hope this doesn't mean there won't be any updates at least until material goes live. Which btw, I've seen screenshots of and it's terrible.

**Comment by brodycas3** [comment]
The new material design is trash. Why do big companies have to ruin good things?

**Comment by jbwzrd213** [comment]
I'm totally sad to hear that you'll probably stop development, but I get it. Do whatever makes you happy.

With that said, I didn't know you had a Patreon until this announcement! I'll be donating, at least until you officially decide to give up the ghost on this. Keep up the great work until then!

**Comment by markreg** [comment]
Is there a way to keep YouTube classic even after Google has pushed their Material Layout permanently? Discard the HTML/JS/CSS sent from Google servers and load a local copy of YouTube classic HTML/JS/CSS?

**Comment by RPGWiZ4RD** [comment]
Very sad news but understandable. Can we at least remove the notification later as I'm primarily using YouTube Plus for the disabling of volume normalization which I expect to continue to work and I can disable other features if they are bugged. So far it's showing the old layout on my primary channel so I'm good to go at least a little longer....

**Comment by mozkan007** [comment]
This is very very awful announcement because I do not think that I can find a magnificent script like this.Youtube is so much beautiful with this versatile script.Even so I appreciate you for your all precious effort.I hope you change your decision and this project does not finish.Thanks for everything and good coding.

**Comment by Enchufadoo** [comment]
Thanks for all your effort anyway.

**Comment by 07416** [comment]
@q1k You have only seen screenshots? For me the material layout beta is automatically enabled every now and then. I remove cookies on a regular basis.

**Comment by hdavidkim** [comment]
Ah... truly a shame. YouTube Plus was definitely the best youtube enhancement script out there. It'll be sorely missed! :(

**Comment by maskkulin** [comment]
we need you, please come back

**Comment by EchoDev** [comment]
Sad to hear you won't be able to work on this addon anymore. 

Thanks for all your work!

**Comment by fengmikaelson** [comment]
the script was deleted on gresyfork？

**Comment by ParticleCore** [comment]
@fengmikaelson please don't use unrelated issues for and use the search next time, that question has already been answered: https://github.com/ParticleCore/Particle/issues/459

**Comment by Jimi-James** [comment]
Thank you very much for keeping the script open-source so that those of us who still need it can download it from a site that DOESN'T have mods/admins who stand by and do nothing while entitled dicks spam undeserved negative reviews. Also thanks for making the script. I hope someone forks or makes their own version soon.

**Comment by 07416** [comment]
@Jimi-James Userscript source is always available.

**Comment by souf99** [comment]
there is one way to keep the extension rolling up is simply by disabling the new youtube layout by scrolling down to most right and click on that little hurry man and simply as that

**Comment by brodycas3** [comment]
@souf99 I don't think you understand why that is there. That's so if you enable the bets that you can opt out of it and go back to normal.

**Comment by ParticleCore** [comment]
@markreg No, the new layout is a complete overhaul of the entire layout, css and js structure. There's absolutely no way to just remove a few things to get the old layout back.

@souf99 Please pay more attention to what was written, this decision has been made for when the new Material layout goes officially live, permanently, with no way to turn it off, not while it is under beta testing.

**Comment by TeddyReight** [comment]
I'm also deeply saddend to see this come to an end. The ad-blocking, disable auto-play on load, and reverse playlist features to name a few, will be sorely missed by me. Good luck with your future endeavors a thank you very much for the time and effort you put into this.

**Comment by willrun4fun** [comment]
Mine had quit working and found my way here.  Sad to hear it will not be moving forward but I totally understand where you are coming from.

**Comment by ParticleCore** [comment]
@willrun4fun If you were using the userscript version hosted in greasyfork then that was just removed for different reasons, you can find the other official hosts for the userscript version here: https://github.com/ParticleCore/Particle/wiki/Download#userscript

**Comment by willrun4fun** [comment]
I am a YouTube Red subscriber and my page has changed design to where the userscript no longer does anything.

**Comment by ParticleCore** [comment]
If that is the case and if it is still possible (which I assume it is since I haven't read about any official YouTube announcement) you can turn opt out of the material design to keep using YT+, if that is something you want: https://github.com/ParticleCore/Particle/wiki/Restore-classic-YouTube

**Comment by Alistair1231** [comment]
Well, I'm glad I stumbled upon this article...
After reading it, I directly made a Patreon account to support you!
You're userscript is the best thing I could ever have wished for. Because of this script I could delete 2 addons and 3 userscripts. 
I really think you should link you're Patreon on the start page of the settings next to this link to get the message out, so all those who love this addon can give you the money for the pizza and coffee you need for programming :D 
Once the material design is unavoidable I would be grateful even for half the features of the current version.
Because then at least not all my youtube customization had to be reaplaced by a bunch of other less nice working and good looking userscripts/addons.
You really deserve some support for all the great work you've done!

**Comment by Jimi-James** [comment]
I had no idea you had a Patreon until just now! I'll definitely add my voice to those who'd be willing to pay/donate for YouTube+ to continue.

**Comment by NovaViper** [comment]
O_o I just loaded my YT and that Material design has arrived!!

**Comment by ParticleCore** [comment]
@NovaViper YT+ should be showing a notice inside a blue box on the bottom right corner when the Material Layout is in use, it also provides a link to the instructions on how to turn off the beta testing of that layout, which you can find here: https://github.com/ParticleCore/Particle/wiki/Restore-classic-YouTube#instructions-for-restoring-classic-youtube

**Comment by NovaViper** [comment]
@ParticleCore Thanks for the tip! But now that the Material update is here.. Does this mean that the plugin will die? I mean, I understand why you would want to (don't get me wrong, I know the feeling as well), but I have grown to highly appreciate this plugin and hope for the best of it and you. Especially that popup player on scroll feature and disable ads feature, those are my favorite ^^

**Comment by ParticleCore** [comment]
@NovaViper please read the announcement with attention, this decision is for when the Material Layout goes live, officially, with no way to turn it off anymore and the decision says "might end" not "will end", so there is a slim chance that I might try to keep development post-material, however, as it is explained in the announcement, that chance is very small.

**Comment by NovaViper** [comment]
@ParticleCore Oh, ok.. Hopefully things will look well for the development. If you need it, I can try to pitch in developing the plugin at the best of my ability

**Comment by PCjabber** [comment]
@ParticleCore, sorry to hear that Google is making things difficult on their end. I really appreciate all your hard work & time spent on this extension though! 

Thank you for the great software, but also for the "heads up" that you might have to abandon it in the future...hopefully that will get others involved in further development.

**Comment by GuardianMajor** [comment]
@ParticleCore I am sorry to hear that the challenges got to be too much but it was expected. Similar issues came about dealing with Facebook and the constant changes and Google is notorious for horrible and crazy ways of doing things with what seems like no rhyme or reason other than to just do them. 

You put up an admirable effort and I hope for the sake of that, you are able to salvage this at some point. Thank you for the greatness while it lasted, seriously Google should try more to implement these features then find ways to break them. Good luck.

**Comment by saintashq** [comment]
I was brought to this by random. But hands down the best addon I used. I wont forget all the help i got from this. Sad, that youtube official doesnt maintain the same level of user friend UI for free users. Thanks for all the efforts and the superb beast of addon. I actually got to know about it after the last update. Feeling sorry to read this but at the same time, cant thank you enough for all the efforts you have put for all the end users. Thank you bro, keep being awesome. Will look forward meeting you again ^^

**Comment by LennyPenny** [comment]
How about hooking the polymer library itself instead of trying to work around the magic it produces?

**Comment by ParticleCore** [comment]
@LennyPenny That was already covered here:

>(...)  the Firefox and Chrome Material layout versions are not the same. YouTube is now using Polymer for the new layout, but due to limitations on Firefox (or just Google trying to push their standards?) this meant that there were two different structures to work with, meaning that YT+ would require a lot more code to avoid having two different versions, each one specific to each browser.

**Comment by LennyPenny** [comment]
I was just assuming that polymer itself would check if the browser is chrome or firefox and then act accordingly, but abstract that away from the youtube devs by them providing a higher level browser independent api, which this project could also make use of 
Now if I follow you correctly they actually send you a different version of polymer itself based on your user agent?

**Comment by birdyhey** [comment]
please no? i can't use youtube without the "blacklist channel" feature. it has completely changed the youtube experience for me. 
It's so easy and efficient to discover new channels thanks to this feature. 

Being able to hide channels that i'm already subbed to, or channels i have watched and didn't like, provides such a clean layout; only showing me the channels/videos i haven't seen, making discovery easy. it's also lightweight, i have thousands of channels hidden at this point,

Is there any chance you would you give another person the reins? 
maybe they could make it work with the new layout. it would be absolutely crushing if it went away, without at least letting someone else try and make it happen somehow!
Thank you!!

**Comment by GuardianMajor** [comment]
@LennyPenny Also keep in mind that, as is always the case with Google, this crap is being rolled out in such a fragmented and atrocious way that unless you are a mind reader, it won't work like you are thinking. For example, I have many Google accounts, currently I can still use this perfectly with 8 of them but the rest that have had the Google crap pushed to them now will switch to the new interface when you log in and bam we get the warning (thanks for that by the way, it helps me know I need to switch accounts) and nothing works.

So if this was actually rolled out to everyone and at the same time, then there would be more of a case for trying to "work with it" as now there is little choice but to work around or against it.

**Comment by Corruptinglyneedful** [comment]
Alternative scripts would be nice.

I can't and refuse to use default Youtube.

First Jeppe's and now this one.

I'm tired of this.

**Comment by ParticleCore** [comment]
@Corruptinglyneedful The pop-up clearly says

>To keep using the current layout without this message please disable YT+

**Comment by vash47** [comment]
Thank you so much for your work until now. I will get on board definitely if you set up a kickstarter or some kind of patreon goal. YT+ is the only thing that makes youtube bearable, despite google trying really heard to fuck it up.

Thanks again.

**Comment by Corruptinglyneedful** [comment]
No, I'm not disabling Youtube Plus.

That would remove all the things that I use Youtube Plus for, like auto quality, disabling auto-play, etc.

That is not a valid solution.  Of course there's going to be no popup if the script is no longer running.

Absolutely not, I am going to continue using Youtube Plus until it no longer works and a viable alternative has surfaced since you are abandoning the project.

I am not just going to disable it because you are abandoning the project and throwing in the towel.

And to continue to leave the popup on the screen as a constant reminder just rubs further salt into that wound.

You should give your users a way to disable that popup after we've acknonwledged the project is shutting down.

Your script doesn't magically cease to exist just because you are throwing in the towel.

**Comment by ParticleCore** [comment]
@vash47 I appreciate yours and all of everyone else's interest, but I was never on this for revenue and I don't intend on giving up yet, this was a very formal and precautionary announcement because it would suck to just one day users not know what the hell happened.

I do understand the frustration of others, I myself felt how hard it was to use YouTube without that power kick YTC offered until it stopped working, it was so annoying that I ended up creating this for myself and shared publicly eventually. I am still looking at YouTube material every time I have a little bit of free time, trying to see a new approach that might work, so it won't be over until it is over.

@Corruptinglyneedful The warning is in place because the rest of the functions are broken or simply do not work, this means that at that point, YT+ is effectively a simple blue box, which is why it would be better to disable it if you wish to keep using the Material layout. Adding an option to dismiss that message would bring nothing to the users' experience since it isn't compatible with the new layout, it would be the same as having it disabled or removed. For these reasons the message box does not offer a way to dismiss it. If you really want to, you can use Stylish, or similar CSS manipulation extension, and hide it, but I will not change it and demanding it will lead to nowhere. However, if you happen to be having the same issue the other users posted here: https://github.com/ParticleCore/Particle/issues/474 then that will be fixed very soon and the message will no longer show up since it is a false detection.

**Comment by Corruptinglyneedful** [comment]
@ParticleCore Ah, I see.  While I get your reasoning, I don't agree with it.

I also don't agree with you throwing in the towel and abandoning us and the project, either.

I think you should absolutely take money if people are going to offer it.  Your script has no equal or alternative right now.  People are going to be looking for one, you could profit from that want and if people are willing to do it, what's the harm?

Just give a few more days and you're going to have throngs of users bitching at your doors.  Once people figure out what's up.

**Comment by LennyPenny** [comment]
@ParticleCore any thoughts on the idea of hooking polymer itself rather than trying to working directly on elements?  https://github.com/ParticleCore/Particle/issues/448#issuecomment-287926425

**Comment by ParticleCore** [comment]
@LennyPenny  It is an approach that I have already tried and was not acceptable, but that doesn't mean I won't eventually try it again.

**Comment by luxoflux** [comment]
I simply block cookies on YouTube which allows me to continue using YT+ with the old layout.

**Comment by Corruptinglyneedful** [comment]
@luxoflux I think you are slightly confused.  The old layout works just fine, it's the new material layout that it won't work at all.

**Comment by luxoflux** [comment]
@Corruptinglyneedful YouTube is pushing the new layout hard as of late, but blocking the cookie from being set works to disable it (for now).

**Comment by Corruptinglyneedful** [comment]
They need to fuck off with that new layout.

**Comment by ddd111ddd** [comment]
Now YouTube is not showing the icon to disable the new layout anymore (confirmed in two different browsers). I had to edit my cookies to disable it.

**Comment by ParticleCore** [comment]
Good news is I'm working on the Material version as of now, I will try to push the alpha version publicly as soon as I can.

Bad news is my time is still very limited, meaning that this process will take time to conclude. Additionally, the entire script will be re-written from scratch and focused on modules, this way it will be far easier for users in the userscript version to just "hack-out" the features they do not want and use just the ones they like. Since this process will be slate zero it means that users should expect the alpha version to break, it will be a normal trend and will help me fix any bugs that I might have missed.

Once a minimally working alpha is ready for public testing I will make an announcement.

**Comment by Jimi-James** [comment]
That's awesome! As long as YouTube+ is sure to come back eventually, I can live with replacing whatever features break with other peoples' scripts where those alternatives exist in the meantime, while I wait for YouTube+ features to come back that will probably never be implemented elsewhere, like seeing a small player in the corner when you scroll down.

**Comment by Corruptinglyneedful** [comment]
This makes me very happy.  I understand your time is valuable.   Please start a patreon and let people donate for development.

I've been looking at other scripts and they do not hold a candle to this one on Firefox.

This makes me very happy indeed.

**Comment by ParticleCore** [comment]
@Corruptinglyneedful I had already set one Patreon page before when other users requested for it. While it is not linked in the extension itself, any and all donation methods are always up to date in the following page: https://github.com/ParticleCore/Particle/wiki/Donate

**Comment by LennyPenny** [comment]
Making it more modular will also make it easier to contribute! 
Will definitely see if I can make some prs once you put it up

**Comment by NovaViper** [comment]
@LennyPenny Yeah and I can try to port it to Vivaldi as much as I can

**Comment by briank6932** [comment]
any idea when you'll have the alpha ready?

**Comment by ParticleCore** [comment]
@briank6932 No. Once it is ready I will announce it.

**Comment by robmonte** [comment]
The first thing I noticed immediately on the new layout is that videos don't resize to the maximum available vertical space in the browser. Would that possible be something you could continue to provide, the custom video size "fit to page max width" option, given they don't end up implementing something similar?

**Comment by brodycas3** [comment]
I agree, That was a stupid choice for them to remove it and I would like it back.

**Comment by ParticleCore** [comment]
Closing, the new version for the YouTube Material Layout has been announced here: #493

**Comment by Sauruman** [comment]
ParticleCore, how would you feel if a Go-Fund-Me page was set-up just for you to continue your work on YT+? When the Go-Fund-Me target reaches high enough, then you can comfortably work on YT+ (Even if it's just Firefox... Chrome sucks anyway ;)

**Comment by ParticleCore** [comment]
@Sauruman The development of the new extension has already begun, you can read more about it here: https://github.com/ParticleCore/Particle/issues/504

**Comment by HimuraMai** [comment]
Is there a way to keep the layout in yt+? Because I can't seem to figure it out at all.

**Comment by ParticleCore** [comment]
@HimuraMai https://github.com/ParticleCore/Particle/wiki/Restore-classic-YouTube

**Comment by GKid94** [comment]
Is there a YT+ for the new Material Design on Chrome?

**Comment by ParticleCore** [comment]
@GKid94 read https://github.com/ParticleCore/Particle/issues/504

**Comment by Corruptinglyneedful** [comment]
Does no one fucking read?  Honestly..

**Comment by GKid94** [comment]
@ParticleCore Thx for making the Iridium extension now I can enjoy the new YouTube layout with some of the YT+ options, I wish you all the luck with Iridium and I hope it will continue to get updated and bring us new options for example the "Remove Autoplay Up Next" and the "Reverse and Autoplay" option for the playlists

**Comment by a0193143** [comment]
I can't click "Restore old Youtube" because the message box cover the button.
Can you make a X button to dismiss the message box?

**Comment by Jimi-James** [comment]
@a0193143 Until that fix comes, here's a tip: you can use your brower's web development menus (well, if it's Firefox or Chrome) to click that message box element with your mouse to select it, and then delete it from the currently loaded page.

**Comment by Eisys** [comment]
Or, the simpler solution, just resize the page.

**Comment by ParticleCore** [comment]
This thread no longer serves a purpose by being left open other than to email spam subscribers, will be closing it to prevent this.
